c10/util/Half.h

```diff
@@ -402,6 +402,8 @@ struct alignas(4) complex<Half> {
   complex() = default;
   // Half constructor is not constexpr so the following constructor can't
   // be constexpr
+  C10_HOST_DEVICE explicit inline complex(const Half& real, const Half& imag)
```

Comment:
New constructor for complex<Half>

Reply:
let's add unit test for this constructor in C++?

Reply:
Do you know if there are C++ tests for complex? I couldn't find them (though there is one testing for CUDA  in `aten/src/ATen/test/cuda_complex_test.cu`) . Should I just create one for complex32 in `aten/src/ATen/test/` for CPU and update the one for CUDA?

Reply:
cc. @zasdfgbnm who added `c10::complex` would know

Reply:
Considering that `complex<Half>` has a different implementation from the `c10::complex` template, my suggestion is to add tests to https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/test/cuda_half_test.cu.

Another option might be to remove this `complex<Half>` specialization and make it the same as complex float and double. But I believe this will need much more work.